python/packages/autogen-core/src/autogen_core/application/_worker_runtime_host_servicer.py

```diff
@@ -68,13 +69,17 @@ async def OpenChannel(  # type: ignore
             for future in self._pending_responses.pop(client_id, {}).values():
                 future.cancel()
             # Remove the client id from the agent type to client id mapping.
-            async with self._agent_type_to_client_id_lock:
-                agent_types = [
-                    agent_type for agent_type, id_ in self._agent_type_to_client_id.items() if id_ == client_id
-                ]
-                for agent_type in agent_types:
-                    del self._agent_type_to_client_id[agent_type]
-            logger.info(f"Client {client_id} disconnected.")
+            await self.on_client_disconnect(client_id)
+
+    async def on_client_disconnect(self, client_id: int) -> None:
+        async with self._agent_type_to_client_id_lock:
+            agent_types = [agent_type for agent_type, id_ in self._agent_type_to_client_id.items() if id_ == client_id]
+            for agent_type in agent_types:
+                del self._agent_type_to_client_id[agent_type]
+            for sub_id in self._client_id_to_subscription_id_mapping.get(client_id, []):
+                logger.info(f"Removing subscription {id}!")
```

Comment:
{id} by itself is not informative. Can we also log the client id? Also the agent type in the removal of the agent types